src/Cello/problem_MethodOrderHilbert.cpp

```diff
@@ -0,0 +1,556 @@
+// See LICENSE_CELLO file for license and copyright information
+
+/// @file     problem_MethodOrderHilbert.cpp
+/// @author   John Brennan (john.brennnan@mu.ie)
+/// @date     2023-11-13
+/// @brief
+
+#include "problem.hpp"
+
+// #define TRACE_ORDER
+
+#ifdef TRACE_ORDER
+#  define TRACE_ORDER_BLOCK(MSG,BLOCK)          \
+  CkPrintf ("TRACE_ORDER %s %s\n",              \
+            std::string(MSG).c_str(),           \
+            BLOCK->name().c_str());             \
+  fflush(stdout);
+#else
+#  define TRACE_ORDER_BLOCK(MSG,BLOCK) /* ... */
+#endif
+
+
+//----------------------------------------------------------------------
+
+MethodOrderHilbert::MethodOrderHilbert(int min_level) throw ()
+  : Method(),
+    is_index_(-1),
+    is_weight_(-1),
+    is_weight_child_(-1),
+    min_level_(min_level)
+{
+  Refresh * refresh = cello::refresh(ir_post_);
+  cello::simulation()->refresh_set_name(ir_post_,name());
+  refresh->add_field("density");
+
+  /// Create Scalar data for ordering index
+  const int n = cello::num_children();
+  is_index_        = cello::scalar_descr_long_long()->new_value(name() + ":index");
+  is_count_        = cello::scalar_descr_long_long()->new_value(name() + ":count");
+  is_next_         = cello::scalar_descr_index()->new_value(name() + ":next");
+  is_weight_       = cello::scalar_descr_long_long()->new_value(name() + ":weight");
+  is_weight_child_ = cello::scalar_descr_long_long()->new_value(name() + ":weight_child",n);
+  is_sync_index_   = cello::scalar_descr_sync()->new_value(name() + ":sync_index");
+  is_sync_weight_  = cello::scalar_descr_sync()->new_value(name() + ":sync_weight");
+}
+
+//======================================================================
+
+void MethodOrderHilbert::compute (Block * block) throw()
+{
+  // Initialize counters, then barrier to ensure counters initialized
+  // before first entry method can arrive
+  TRACE_ORDER_BLOCK("compute",block);
+  Sync * sync_index = psync_index_(block);
+  Sync * sync_weight = psync_weight_(block);
+
+  *pindex_(block) = 0;
+  *pcount_(block) = 0;
+  *pweight_(block) = 1;
```

Comment:
Are all blocks equally weighted?  If so, this is fine.  It's just for my curiosity.  If we were to have performance timing for each block or some proxy, would this be where the associated weight would be assigned?

Reply:
Yeah, the weighting is the same as in the Morton method as @jobordner said. So all the blocks initially have equal weight and by the end of the method all leaf blocks should have weight 1 and all non-leaf blocks should have a weight = 1 + (the number of its descendant blocks)